src/components/alertComponent/alert.tsx

```diff
@@ -14,11 +15,21 @@ const Alert: React.FC = () => {
       <Badge
         badgeContent={alert.newAlerts ? alert.newAlerts.length : 0}
         color="primary"
-        onClick={e => {
+        onClick={async e => {
           e.stopPropagation();
-          setOpen(!open);
-          if (alert.newAlerts && alert.newAlerts.length) {
+          await setOpen(!open);
+          if (alert.newAlerts && alert.newAlerts.length && open) {
             dispatch(clearAlerts(alert.newAlerts, user.id));
+          } else if (!open) {
+            if (alert.alerts.length) {
+              return new Promise(res => {
+                res(
+                  alert.alerts.forEach(async curAlert => {
+                    await axios.delete(`/api/alert/${curAlert.id}`);
```

Comment:
would be better to send all of the alerts at once and handle the data on the back end. this way makes excessive calls to the server when one would do.

Reply:
delete requests cannot send a req.body